Formula/mesos.rb

```diff
@@ -53,6 +53,16 @@ class Mesos < Formula
   def install
     ENV.java_cache
 
+    # Disable optimizing as libc++ does not play well with optimized clang
+    # builds (see https://llvm.org/bugs/show_bug.cgi?id=28469 and
+    # https://issues.apache.org/jira/browse/MESOS-5745).
+    #
+    # NOTE: We cannot use `--disable-optimize` since we also pass e.g.,
+    # CXXFLAGS via environment variables. Since compiler flags are passed via
+    # environment variables the Mesos build will silently ignore flags like
+    # `--[disable|enable]-optimize`.
+    ENV.O0
```

Comment:
May as well pass `--disable-optimize` too. Also, you can guard this with something like `unless DevelopmentTools.clang_build_version >= 900`